src/test/enhetstester/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vilkårsvurdering/VilkårsvurderingStegTest.kt

```diff
@@ -0,0 +1,235 @@
+        val søkerPersonResultat = PersonResultat(vilkårsvurdering = vilkårsvurderingForSøker, aktør = søker)
+        val barnPersonResultat = PersonResultat(vilkårsvurdering = vilkårsvurderingForSøker, aktør = barn)
+
+        søkerPersonResultat.setSortedVilkårResultater(
+            setOf(
+                VilkårResultat(
+                    personResultat = søkerPersonResultat,
+                    vilkårType = Vilkår.BOSATT_I_RIKET,
+                    resultat = Resultat.OPPFYLT,
+                    periodeFom = LocalDate.of(2018, 12, 12),
+                    periodeTom = LocalDate.of(2022, 12, 12),
+                    begrunnelse = "",
+                    behandlingId = behandling.id
+                ),
+            )
+        )
+
+        barnPersonResultat.setSortedVilkårResultater(
+            setOf(
+                VilkårResultat(
+                    personResultat = søkerPersonResultat,
+                    vilkårType = Vilkår.BARNEHAGEPLASS,
+                    resultat = Resultat.OPPFYLT,
+                    periodeFom = LocalDate.of(2018, 12, 12),
+                    periodeTom = LocalDate.of(2022, 12, 12),
+                    begrunnelse = "",
+                    behandlingId = behandling.id,
+                    antallTimer = BigDecimal(25)
+                ),
+                VilkårResultat(
+                    personResultat = søkerPersonResultat,
+                    vilkårType = Vilkår.BOR_MED_SØKER,
+                    resultat = Resultat.OPPFYLT,
+                    utdypendeVilkårsvurderinger = listOf(UtdypendeVilkårsvurdering.DELT_BOSTED),
+                    periodeFom = LocalDate.of(2020, 12, 12),
+                    periodeTom = LocalDate.of(2022, 12, 12),
+                    begrunnelse = "",
+                    behandlingId = behandling.id
+                )
+            )
+        )
+
+        vilkårsvurderingForSøker.personResultater = setOf(søkerPersonResultat, barnPersonResultat)
+
+        every { behandlingService.hentBehandling(behandling.id) } returns behandling
+        every { personopplysningGrunnlagService.hentAktivPersonopplysningGrunnlagThrows(any()) } returns personopplysningGrunnlag
+        every { vilkårsvurderingService.hentAktivVilkårsvurderingForBehandling(behandling.id) } returns vilkårsvurderingForSøker
+
+        val feil = assertThrows<FunksjonellFeil> {
+            vilkårsvurderingSteg.utførSteg(behandling.id)
+        }
+
+        assertThat(
+            feil.message,
+            Is("Det er lagt inn gradert barnehageplass og delt bosted for samme periode.")
+        )
+
+        verify(exactly = 1) { behandlingService.hentBehandling(behandling.id) }
+        verify(exactly = 1) { personopplysningGrunnlagService.hentAktivPersonopplysningGrunnlagThrows(any()) }
+        verify(exactly = 1) { vilkårsvurderingService.hentAktivVilkårsvurderingForBehandling(behandling.id) }
+    }
+
+    @Test
+    fun `utførSteg - skal kaste validere vilkårsvurderingen`() {
```

Comment:
Liten typo her. Skal vel være noe sånt som:
```suggestion
    fun `utførSteg - skal validere vilkårsvurderingen`() {
```

Reply:
Haha yikes 🙈